src/openvic-simulation/GameRulesManager.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "openvic-simulation/utility/Getters.hpp"
+namespace OpenVic {
+	struct GameRulesManager {
+	private:
+		bool PROPERTY_RW(use_legacy_farm_mine_logic);
+		bool PROPERTY_RW(use_legacy_ui);
```

Comment:
```suggestion
		bool PROPERTY_RW(use_legacy_farm_mine_logic, true);
		bool PROPERTY_RW(use_legacy_ui, true);
```
- You need to explicitly initialise the bools, otherwise they'll have random values with MSVC. If you rebase onto master you can use the new `PROPERTY` initialisation value feature Spartan added, as shown above (adding the starting value as another argument in the macro).
- Using the `PROPERTY` initialisation means we won't need a constructor, so you can remove `GameRulesManager();` from this file and delete `GameRulesManager.cpp` altogether. Also maybe move this to the `misc` folder? It fits with general infrastructure stuff like `SimulationClock` in there.
- I'm not sure what `use_legacy_ui` is meant to be for, I expect we'd use rules for specific features rather than general "legacy UI" vs "new UI"
- IMO the rules should enable new/divergent features, rather than enabling legacy stuff. I expect there'll be rules with more options than just true/false, which means there won't be a single non-legacy baseline. Better to have "faithful" as the baseline/default and have rules to enable specific divergent behaviour.